src/main.c

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <math.h>
+#include "../include/randomizer_8_16_64.h"
+
+void show_input(int **input){
+    
+    for(int i = 0, exp = 3; i < 3; i++){
+        
+        exp += i;
+        for(int j = 0; j < pow(2,exp);j++){
```

Comment:
Das concept von 2^exp Zeile: 9 ist folgender: 
Da die Arrays die Größen 8, 16, 64 haben und die einzelnen array über input[0], input[1], input[2] zugreifbar sind, ermöglicht die Form 2^(exp + i) mit exp = 3, die ausgabe 8, 16, 64 -> aus diesem Grund ist die Abbruchbedingung der ineren Schleife 2^(exp+i)

**i = 0; |exp = 3;**
2 ^ (exp + i) = 2^3 = 8

**i = 1| exp = exp + i = 3 + 1**
2 ^ (exp + i) = 2^4 = 16

**i = 2 | exp = exp + i = 4 + 2** 
2 ^ (exp + i) = 2 ^ 6 = 64

Reply:
Klingt gut.
Wir werden ja für den Leistungsvergleich auch noch andere sizes brauchen (lt Angabe), fraglich daher für mich wie das ggf. auch weiterverwendet werden könnte... aber derzeit top!

Reply:
Hast völlig recht! Für den Leistungsvergleich für die folgenden Größen 8, 32, 128, 512, 2048, etc. habe ich mir einen ähnlichen Vorgang überlegt:
`for(int i = 0, exp = 3; i < 8; i++){
    exp = exp + 2;
    for(int j = 0; j < pow(2,exp); j++){...}
}`
da 8, 32, 128, etc auch als 2 ^ 3, 2^5, 2^7, 2^9, etc ausgedrückt werden können, müssen wir hier nur den exponent statt mit i immer um 2 inkrementieren und die abbruchbedingung der äußeren For Schleife auf 8 ändern da wir 8 arrays haben. -> So könnten wir ziemlich dynamisch alle arrays durch itterieren / reservieren / freigeben